### PR TITLE
fix(argo-cd): remove secretName for server and applicationSet Certificates

### PR DESCRIPTION
**Pull request by erwanval**
argocd-server and argocd-applicationset are expecting a secret with a predefined name, so having the option to set a custom name for the secret is confusing.

Related Issue:
#2724

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by jmeridth** [comment]
The default values that are assigned match the default values.  You do not need to have them in your values.yaml and the default values will be set.  I'm not a fan of this PR but await feedback from other maintainers.

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -26,5 +26,7 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: fixed
-      description: Fixed redis exporter NOAUTH error
+    - kind: removed
+      description: Remove server.certificate.serverName, as the expected secret name is static (argocd-server-tls)
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:30]:
```suggestion
      description: Remove `server.certificate.secretName`, as the expected secret name is static (argocd-server-tls)
```

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -26,5 +26,7 @@ annotations:
...
+    - kind: removed
+      description: Remove applicationSet.certificate.serverName, as the expected secret name is static (argocd-applicationset-controller-tls)
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:32]:
```suggestion
      description: Remove `applicationSet.certificate.secretName`, as the expected secret name is static (argocd-applicationset-controller-tls)
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
I have no strict opinion here. I am fine with both.

But the changelog is wrong (you mentioned the wrong parameters).

**Comment by jmeridth** [comment]
Closing and reopening to see if DCO kicks off (was off and is on again)

**Comment by jmeridth** [comment]
Queue might be backed up still due to https://github.com/dcoapp/app/issues/211.  Can try again later.

**Comment by jmeridth** [comment]
@erwanval when you make the changes requested by @mkilchhofer, maybe a new commit will trigger the DCO app.  We'll see.

**Comment by erwanval** [comment]
> The default values that are assigned match the default values. You do not need to have them in your values.yaml and the default values will be set. I'm not a fan of this PR but await feedback from other maintainers.

If you keep the default, it works indeed. However, any other values than the default will be ignored, and the application will generate its own self-signed certificate instead. It still starts without error in appearance, but can cause confusion since the exposed cert doesn't match the one you configured.

@mkilchhofer Thanks for fixing the typo in changelog.
Regarding DCO, it seems to have been triggered now, but there is no "Verified" next to my commits somehow? I'm not sure if I'm missing something, it usually works just fine with other repositories.

**Comment by mkilchhofer** [comment]
> Regarding DCO, it seems to have been triggered now, but there is no "Verified" next to my commits somehow? I'm not sure if I'm missing something, it usually works just fine with other repositories.

DCO is perfectly fine now (green rectangles). It was failing as the DCO GitHub app was not okay.
image

(The verified commits (gpg key or via GitHub GUI) are not required in argo-helm) :)

**Review by mkilchhofer — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

**Comment by yu-croco** [comment]
Hi @erwanval , can you please resolve conflict?

**Comment by erwanval** [comment]
@yu-croco done, thanks

**Review by yu-croco — APPROVED** [review]

**Pull request merged**